plutus-core/plutus-core/test/Evaluation/Spec.hs

```diff
@@ -1,11 +1,247 @@
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+
 module Evaluation.Spec where
 
+import Control.Monad.Except
+import Data.ByteString qualified as BS
+import Data.Functor ((<&>))
+import Data.Int (Int64)
+import Data.List.Extra qualified as List
+import Data.Text (Text)
+import Data.Type.Equality
+import Data.Typeable (splitTyConApp)
 import Evaluation.Machines (test_machines)
-
+import Hedgehog hiding (Opaque, Var, eval)
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import PlutusCore hiding (Term)
+import PlutusCore qualified as PLC
+import PlutusCore.Builtin
+import PlutusCore.Data (Data (..))
+import PlutusCore.Generators
+import Prettyprinter
 import Test.Tasty
+import Test.Tasty.Hedgehog
+import Type.Reflection
+import Unsafe.Coerce
+
+type Term = PLC.Term TyName Name DefaultUni DefaultFun ()
+
+test_builtinsDon'tThrow :: TestTree
+test_builtinsDon'tThrow =
+  testGroup
+    "Builtins don't throw"
+    . fmap (\fun -> testProperty (show fun) $ prop_builtinsDon'tThrow fun)
+    $ List.enumerate
+      -- TODO: remove this
+      List.\\ [ VerifySignature,
+                VerifyEcdsaSecp256k1Signature,
+                VerifySchnorrSecp256k1Signature
+              ]
+
+prop_builtinsDon'tThrow :: DefaultFun -> Property
+prop_builtinsDon'tThrow bn = property $ do
+  (args, argStrings) <- unzip <$> forAllNoShow (genArgs bn)
+  let (res, logs) = runEmitter . runExceptT $ eval args argStrings
+  case res of
+    Right _ -> success
+    Left err -> do
+      liftIO $ do
+        putStrLn "Builtin function evaluation failed"
+        putStrLn $ "Function: " <> show bn
+        putStrLn $ "Arguments: " <> show argStrings
+        putStrLn $
+          "Error: "
+            <> ( case err of
+                   KnownTypeEvaluationFailure -> "KnownTypeEvaluationFailure"
+                   KnownTypeUnliftingError e ->
+                     "KnownTypeUnliftingError: "
+                       <> show (pretty e)
+               )
+        putStrLn $ "Execution log: " <> show logs
+      failure
+  where
+    meaning :: BuiltinMeaning Term (CostingPart DefaultUni DefaultFun)
+    meaning = toBuiltinMeaning bn
+
+    eval :: [Term] -> [String] -> MakeKnownM Term
+    eval args0 argStrings = case meaning of
+      BuiltinMeaning _ _ runtime -> go (_broRuntimeScheme runtime) (_broImmediateF runtime) args0
+      where
+        go ::
+          forall n.
+          RuntimeScheme n ->
+          ToRuntimeDenotationType Term n ->
+          [Term] ->
+          MakeKnownM Term
+        go sch f args = case (sch, args) of
+          (RuntimeSchemeArrow sch', a : as) -> do
+            res <- liftEither (f a)
+            go sch' res as
+          (RuntimeSchemeResult, []) -> f
+          (RuntimeSchemeAll sch', _) -> go sch' f args
+          _ -> error $ "Wrong number of args for builtin " <> show bn <> ": " <> show argStrings
```

Comment:
You could probably make this function run in `GenT MakeKnownM` and generate arguments on the fly to avoid this `error` case, but it's perfectly fine this way too.

Reply:
Not immediately obvious how to do this, so I added a TODO above this line.